tests/TestCase/Controller/Component/ModulesComponentTest.php

```diff
@@ -155,6 +156,101 @@ public function testGetProject($expected, $meta) : void
         static::assertEquals($expected, $actual);
     }
 
+    /**
+     * Data provider for `testIsAbstract` test case.
+     *
+     * @return array
+     */
+    public function isAbstractProvider() : array
+    {
+        return [
+            'isAbstractTrue' => [
+                true,
+                'objects',
+            ],
+            'isAbstractFalse' => [
+                false,
+                'documents',
+            ],
+        ];
+    }
+
+    /**
+     * Test `isAbstract()` method.
+     *
+     * @dataProvider isAbstractProvider();
+     * @covers ::isAbstract()
+     *
+     * @return void
+     */
+    public function testIsAbstract($expected, $data) : void
```

Comment:
Add tests and coverage on `ModulesComponent` is a good idea, but better do it in a different PR
These methods are not related to the current issue

Reply:
Ok, I will separate it